### PR TITLE
Cleanup fixes for tests waiting for DVs to succeed

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This is a follow-up to https://github.com/kubevirt/ssp-operator/pull/418

1. Do not check PVC pod annotation because it might not be always present (i.e. when smartcloning).
2. Reenable CDI garbage collection in `dataSources_test.go`
3. Unquarantine no longer failing tests in `dataSources_test.go`

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
/hold

We need to wait until https://github.com/kubevirt/containerized-data-importer/pull/2441 or https://github.com/kubevirt/containerized-data-importer/pull/2452 is available in a release.

**Comment by 0xFelix** [comment]
[CDI v1.55.1](https://github.com/kubevirt/containerized-data-importer/releases/tag/v1.55.1) was released.

File: tests/dataSources_test.go [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ var _ = Describe("DataSources", func() {
 					}, env.Timeout(), time.Second).Should(Equal(metav1.StatusReasonNotFound), "DataImportCron should not exist.")
 				})
 
-				PIt("[QUARANTINE][test_id:8298] should restore DataSource, when CDI label is removed", func() {
+				It("t[test_id:8298] should restore DataSource, when CDI label is removed", func() {
```
Comment (akrejcir) [inline tests/dataSources_test.go:1137]: a typo

**Pull request merged**